frontend/README.md

```diff
@@ -2,28 +2,77 @@
 
 A single-page app for the [OpenCraft Instance Manager (OCIM)](https://github.com/open-craft/opencraft).
 
-# Provisioning
+## Provisioning
 
-1. Install the API client:
-```
+- Install the API client:
+
+```bash
 ./scripts/build-api-client.sh
 ```
-2. Install requirements:
-```
+
+- Install requirements:
+
+```bash
 npm install
 ```
-3. Run frontend:
-```
+
+- Run frontend:
+
+```bash
 npm start
 ```
 
-# Running with Vagrant
+- Generating New Components:
+
+```bash
+npm run generate # uses plop, more information here(https://plopjs.com/)
+```
+
+## API Client Instructions
+
+- Updating the API Client:
+
+```bash
+cd ./frontend # go to frontend directory
+npm run update-api-client
+```
+
+- Building the API Client:
+
+```bash
+cd ./frontend # go to frontend directory
+npm run build-api-client
+```
+
+## Testing
+
+- Running the tests:
+
+```bash
+npm run lint # for running the linting
+
+npm run lint-fix # for fixing the linting
+
+npm run test # for running the tests
+
+make test.js && make test.frontend && npm run lint # for running the test using the makefile
+```
+
+## Deployment Process
+
+The Deployment been done on the master automatically using CircleCI(for details, check `frontend-deploy` job in [circle.yml](../circle.yml)). The bundle been hosted on S3 and been directly served from the Cloudfront.
+
+## Running with Vagrant
 
 While it is possible to run this frontend server within Vagrant during
 development, for performance reasons it's better to run it separately
 outside of Vagrant instead.
 
-# Frontend Architecture/Stack
+## Reuseable UI Components
+
+All reusable UI components are described in the `/demo` route, accessible in development environment through [localhost](http://localhost:3000/demo).
```

Comment:
Not all components are listed there.
```suggestion
A partial list of reusable UI components is shown in the `/demo` route, accessible in the development environment through [http://localhost:3000/demo](http://localhost:3000/demo).
```

Reply:
Thanks for the suggestion, I have added it.